File: IN1000/Obliger/Uke 10 og 11 peasent/rutenett.py
from random import randint
from celle import Celle

class Rutenett:
    def __init__(self, rader, kolonner):
        self._ant_rader = rader
        self._ant_kolonner = kolonner
        self._rutenett = self._lag_tomt_rutenett()

    def _lag_tomt_rutenett(self):
        noested_liste = []
        for i in range(self._ant_rader):
            noested_liste.append(self._lag_tom_rad())
        return noested_liste

    def _lag_tom_rad(self):
        liste = []
        for i in range(self._ant_kolonner):
            liste.append(None)
        return liste

    def fyll_med_tilfeldige_celler(self):
        for y in range(len(self._rutenett)):
            for x in range(len(self._rutenett[y])):
                self.lag_celle(y, x)

    def lag_celle(self, rad, kol):
        celle = Celle()
        self._rutenett[rad][kol] = celle
        if randint(0,2) == 0:
            celle.sett_levende()

    def hent_celle(self, rad, kol):
        if rad < 0 or rad > len(self._rutenett)-1 or kol < 0 or kol > len(self._rutenett[rad])-1:
            return None
        return self._rutenett[rad][kol]

    def tegn_rutenett(self):
        for i in range(10):
            print()
        for y in range(len(self._rutenett)):
            for x in range(len(self._rutenett[y])):
                print(self._rutenett[y][x].hent_status_tegn(), end = "")
            print("\n", end = "")

    def _sett_naboer(self, rad, kol):
        hoved_celle = self.hent_celle(rad, kol)
        for y in range(-1,2):
            for x in range(-1,2):
                if (rad + y, kol + x) != (rad, kol):
                    if self.hent_celle(rad + y, kol + x) != None:
                        hoved_celle.legg_til_nabo(self.hent_celle(rad + y, kol + x))

    def koble_celler(self):
        for y in range(len(self._rutenett)):
            for x in range(len(self._rutenett[y])):
                self._sett_naboer(y, x)

    def hent_alle_celler(self):
        liste = []
        for y in range(len(self._rutenett)):
            for x in range(len(self._rutenett)):
                liste.append(self.hent_celle(y, x))
        return liste
                

    def antall_levende(self):
        antall_levende = 0
        for celle in self.hent_alle_celler():
            if celle.er_levende():
                antall_levende += 1
        return antall_levende

File: IN1010/Obliger/Oblig1/Rutenett.java
import java.util.Random;
import java.lang.Math;

public class Rutenett {
    int antRader;
    int antKolonner;
    Celle[][] rutene;

    public Rutenett(int antRader, int antKolonner) {
        this.antRader = antRader;
        this.antKolonner = antKolonner;
        rutene = new Celle[antRader][antKolonner];
    }

    public void lagCelle(int rad, int kol) {
        Celle celle = new Celle();
        Random random = new Random();
        if (random.nextInt(0, 3) == 0) {
            celle.settLevende();
        }
        rutene[rad][kol] = celle;
    }

    public void fyllMedTilfeldigeCeller() {
        for (int x = 0; x < antRader; x++) {
            for (int y = 0; y < antKolonner; y++) {
                this.lagCelle(x, y);
            }
        }
    }

    public Celle hentCelle(int rad, int kol) {
        if (rad > antRader - 1 || kol > antKolonner - 1 || rad < 0 || kol < 0) {
            return null;
        } else {
            return rutene[rad][kol];
        }
    }

    public void tegnRutenett() {
        for (int x = 0; x < antRader; x++) {
            System.out.println();
            for (int i = 0; i < antKolonner; i++) {
                System.out.print("+---");
            } System.out.print("+");
            System.out.println();
            System.out.print("|");
            for (int y = 0; y < antKolonner; y++) {
                System.out.print(" " + rutene[x][y].hentStatusTegn() + " |");
            }
        }
        System.out.println();
        for (int i = 0; i < antKolonner; i++) {
            System.out.print("+---");
        } System.out.print("+");
    }

    public void settNaboer(int rad, int kol) {
        for (int x = -1; x < 2; x++) {
            for (int y = -1; y < 2; y++) {
                if (Math.abs(x) + Math.abs(y) != 0) {
                    rutene[rad][kol].leggTilNabo(this.hentCelle(rad + x, kol + y));
                }
            }
        }
    }

    public void kobleAlleCeller() {
        for (int x = 0; x < antRader; x++) {
            for (int y = 0; y < antKolonner; y++) {
                this.settNaboer(x, y);
            }
        }
    }

    public int antallLevende() {
        int teller = 0;
        for (int x = 0; x < antRader; x++) {
            for (int y = 0; y < antKolonner; y++) {
                if (rutene[x][y].erLevende()) {
                    teller++;
                }
            }
        }
        return teller;
    }

    public void hjelp() {
        for (int x = 0; x < antRader; x++) {
            for (int y = 0; y < antKolonner; y++) {
                rutene[x][y].tellLevendeNaboer();
            }
        }
    }
}
